ui/src/chatbot.tsx

```diff
@@ -134,7 +143,22 @@ export const XChatbot = (props: Chatbot) => {
         setIsInfiniteLoading(true)
       }
     }, [props.events, props.name]),
-    onDataChange = React.useCallback(() => { if (props.data) setMsgs(processData(props.data)) }, [props.data])
+    onDataChange = React.useCallback(() => { if (props.data) setMsgs(processData(props.data)) }, [props.data]),
+    handleFeedback = (positive: B, id: I) => {
+      if (props.events?.includes('feedback')) {
+        setRated(rated => {
+          if (rated.has(id) && rated.get(id) === positive) {
```

Comment:
Wouldn't this be enough?
```suggestion
          if (rated.get(id) === positive) {
```